src/kirby/components/modal/modal.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component } from '@angular/core';
+import { NavParams, ModalController } from '@ionic/angular';
+
+import { ModalConfig } from './config/modal-config';
+import { ModalConfigHelper } from './helpers/modal-config-helper';
+
+@Component({
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss'],
+})
+export class ModalComponent {
+  config: ModalConfig;
+
+  constructor(private modalController: ModalController, params: NavParams) {
+    this.config = ModalConfigHelper.processOptionalValues(params.get('config'));
+  }
+
+  dismissModal(): void {
+    this.modalController.dismiss();
```

Comment:
Shouldn't this be handled by the ModalService to make it consistent with {N} implementation? What if we have some additional logic, guards, events in the service when closing?

Reply:
I completely agree and I wanted to call the ModalService too, but there is a circular dependency, which prevents me from doing that. I will put a todo on it, but i'd say let's keep it as it is for the skeleton version.

Reply:
Despite that, additional logic when closing, such as guards and events should be handled inside modal.onDidDismiss() and inside the then((data) => ...) Promises. This is also invoked when closing the modal via ModalComponent's methods, as well as when closing the modal by clicking outside of it (in the shadow). The ModalService.hideModal() should be only a "proxy", which enables access to these promises for the outside (e.g. nested components, or the caller component) and it should not hide any additional logic